src/app/services/schema/app-schema.service.ts

```diff
@@ -637,65 +639,74 @@ export class AppSchemaService {
       }));
   }
 
-  private handleSchemaSubQuestions(payload: CommonSchemaAddControl, newFormControl: CustomUntypedFormControl): void {
-    const {
-      schema, isNew, path, subscription, formGroup, config, isParentImmutable,
-    } = payload;
+  private handleSchemaSubQuestions({
+    schema,
+    isNew,
+    path,
+    subscription,
+    formGroup,
+    config,
+    isParentImmutable,
+  }: CommonSchemaAddControl, newFormControl: CustomUntypedFormControl): void {
+    if (!schema.subquestions) {
+      return;
+    }
 
-    if (schema.subquestions) {
-      schema.subquestions.forEach((subquestion) => {
-        subscription.add(
-          this.addFormControls({
-            isNew,
-            path,
-            chartSchemaNode: subquestion,
-            formGroup,
-            config,
-            isParentImmutable: !!schema.immutable || isParentImmutable,
-          }),
-        );
+    for (const subquestion of schema.subquestions) {
+      subscription.add(
+        this.addFormControls({
+          isNew,
+          path,
+          chartSchemaNode: subquestion,
+          formGroup,
+          config,
+          isParentImmutable: !!schema.immutable || isParentImmutable,
+        }),
+      );
 
-        const formField = (formGroup.controls[subquestion.variable] as CustomUntypedFormField);
-        if (!formField.hidden$) {
-          formField.hidden$ = new BehaviorSubject<boolean>(false);
-        }
-        if (newFormControl.value === schema.show_subquestions_if) {
-          formField.hidden$.next(false);
-          formField.enable();
-        } else {
-          formField.hidden$.next(true);
-          formField.disable();
+      const formField = formGroup.controls[subquestion.variable] as CustomUntypedFormField;
+      this.initializeFormField(formField, newFormControl.value, schema, subquestion, isNew, isParentImmutable);
+    }
+
+    subscription.add(newFormControl.valueChanges.subscribe((value) => {
+      for (const subquestion of schema.subquestions) {
+        const parentControl = formGroup.controls[subquestion.variable].parent as CustomUntypedFormField;
+        if (!parentControl.hidden$) {
+          parentControl.hidden$ = new BehaviorSubject<boolean>(false);
         }
-      });
 
-      subscription.add(newFormControl.valueChanges.subscribe((value) => {
-        schema.subquestions.forEach((subquestion) => {
-          const parentControl = (formGroup.controls[subquestion.variable].parent as CustomUntypedFormField);
-          if (!parentControl.hidden$) {
-            parentControl.hidden$ = new BehaviorSubject<boolean>(false);
+        parentControl.hidden$.pipe(take(1)).subscribe((isParentHidden) => {
```

Comment:
See if possible to have a variable `hidden` on the `parentControl` as well which stays update to date with the `hidden$` observable. That way we don't have to rely on a subscription here to make subsequent recursion calls.